test/parallel/test-http2-server-set-invalid-header.js

```diff
@@ -0,0 +1,56 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http2 = require('http2');
+const body =
+  'Cannot set HTTP/2 pseudo-headers';
+
+const invalidHeaders = [
+  { key: ':status', value: 200 }, 
+  { key: ':method', value: 'GET' }, 
+  { key: ':path', value: '/' }, 
+  { key: ':authority', value: 'example.com' }, 
+  { key: ':scheme', value: 'http' }
+];
+
+const checkServer = (header, value) => {
+  const server = http2.createServer((req, res) => {
+    try {
+      res.setHeader('foobar', 'baz');
+      res.setHeader('X-POWERED-BY', 'node-test');
+      // invalid header
+      res.setHeader(header, value);
+    } catch (e) {
+      res.statusCode = 500;
+      res.end(e.message);
+    }
+  });
+  
+  server.listen(0, common.mustCall(() => {
+    const client = http2.connect(`http://localhost:${server.address().port}`);
+    const headers = { ':path': '/' };
+    const req = client.request(headers);
+    req.on('response', common.mustCall((headers) => {
+      assert.strictEqual(headers[':status'], 500);
+      assert.strictEqual(headers['foobar'], 'baz');
+      assert.strictEqual(headers['x-powered-by'], 'node-test');
+    }));
+  
+    let data = '';
+    req.on('data', (d) => data += d);
```

Comment:
https://github.com/nodejs/node/commit/d5158574c6cb69204d155b6ea0ee4c07ab07ddd6 is very old. So, no there should be something else into play here.